src/transformers/pipelines.py

```diff
@@ -432,14 +431,18 @@ def inputs_for_model(self, features: Union[dict, List[dict]]) -> Dict:
         else:
             return {k: [feature[k] for feature in features] for k in args}
 
-    def _parse_and_tokenize(self, *texts, **kwargs):
+    def _parse_and_tokenize(self, *texts, pad_to_max_length=False, **kwargs):
         """
         Parse arguments and tokenize
         """
         # Parse arguments
         inputs = self._args_parser(*texts, **kwargs)
         inputs = self.tokenizer.batch_encode_plus(
-            inputs, add_special_tokens=True, return_tensors=self.framework, max_length=self.tokenizer.max_len
+            inputs,
+            add_special_tokens=True,
+            return_tensors=self.framework,
+            max_length=self.tokenizer.max_len,
+            pad_to_max_length=pad_to_max_length,
```

Comment:
Is there a reason why we would not optionally use `pad_to_max_length` here? Was not sure if I can add it, but for summarization when using a  batched input, this is necessary. @LysandreJik @thomwolf @mfuntowicz